src/webgpu/api/validation/encoding/cmds/render/draw.spec.ts

```diff
@@ -592,17 +611,34 @@ buffer slot and index buffer will cause no validation error, with completely/par
     u //
       .combine('drawType', ['draw', 'drawIndexed', 'drawIndirect', 'drawIndexedIndirect'] as const)
       .beginSubcases()
-      .combine('vertexBoundOffestFactor', [0, 0.5, 1, 1.5, 2])
-      .combine('instanceBoundOffestFactor', [0, 0.5, 1, 1.5, 2])
-      .combine('indexBoundOffestFactor', [0, 0.5, 1, 1.5, 2])
       .combine('arrayStrideState', ['zero', 'exact', 'oversize'] as const)
+      .combineWithParams([
+        { vertexOffsetFactor: 0.0, instanceOffsetFactor: 0.0, indexOffsetFactor: 0.0 },
+        { vertexOffsetFactor: 1.0, instanceOffsetFactor: 1.0, indexOffsetFactor: 1.0 },
+        //
```

Comment:
nit. Since you already included comment sections to separate the cases, maybe a brief one-liner explaining what the cases are testing?

Reply:
Honestly I don't have much of an explanation here. I just picked some reasonable looking subset of the original combinatorial explosion that would maybe have the right amounts of overlaps. But I don't fully understand how they're used so I can't really write intelligently here.

Reply:
Initially these factors are designed to emulate all overlay possibility of three buffer binding within a single GPUBuffer. Consider each binding size is `1`, and the offset factor indicate the begin point of each binding form `0` to `2` in `0.5` steps.
So some of the cases are: 
1. No overlay: [0, 1, 2]
2. The first two binding no overlay, third binding overlay: [0, 1, (1, 1.5)]
3. The first two partially overlay, and the third one may or may not overlay with them: [0, 0.5, (0.5, 1, 1.5)]
4. The first two complete overlay, and the third one may or may not overlay with them: [0, 0, (0, 0.5, 1)]

And three different kind of binding (vertex, instance, index) map to these three binding in several ways.
Yes there are too many combination and many of them is duplicated...

And oops I realize that the expected binding size of three kind is not necessary the same, so combining the factor doesn't ensure all overlay cases are tested.
Maybe I should review this tests later and replace these factor things with explicit overlay expectation...

Reply:
Thanks for the explanation, that's about what I thought but I wasn't sure.
Let me see if we can do something like:

* No overlay: [0, 2, 4]
* The first two binding no overlay, third binding overlay: [0, 2, (2, 2.5)]
* The first two partially overlay, and the third one may or may not overlay with them: [0, 0.5, (0.5, 1, 4)]
* The first two complete overlay, and the third one may or may not overlay with them: [0, 0, (0, 0.5, 4)]

Reply:
Actually - this test is only a tiny fraction of the slowness here, so I'm just going to revert this, add a TODO for what you suggested, and land this PR.